### ISSUE TITLE
GPT2 Models Merge and Unload

### ISSUE THREAD

**Issue by syngokhan** [huggingface/peft#365]
Hello again, do you think about merging for gpt2 models? It would be great if you could do it. When we can't test new models (Alpaca etc), we have to use the old ones (GPT-2). Do you have another method ? I wish you a good work

```

from peft import PeftModel, PeftConfig

peft_model_id = "/content/PEFT/V_1.0/"

config = PeftConfig.from_pretrained(peft_model_id)

tokenizer = AutoTokenizer.from_pretrained(config.base_model_name_or_path)

model = AutoModelWithLMHead.from_pretrained(
    config.base_model_name_or_path, 
    #torchscript = True,
    #load_in_8bit=True,
    #device_map = "auto",
    use_cache = False
)
print("Base Model : ", model.device)

device = "cuda:0" if torch.cuda.is_available() else "cpu"
device = "cpu"

model = PeftModel.from_pretrained(model, peft_model_id)
print("Peft Model : " ,model.device)

if torch.__version__.startswith("2.0"): 
    print("Model Compile Using...")   
    model_comp = torch.compile(model)

model = model.merge_and_unload()

│ in <cell line: 1>:1                                                                              │
│                                                                                                  │
│ /usr/local/lib/python3.9/dist-packages/peft/tuners/lora.py:305 in merge_and_unload               │
│                                                                                                  │
│   302 │   │   as a standalone model.                                                             │
│   303 │   │   """                                                                                │
│   304 │   │   if getattr(self.config, "model_type", None) == "gpt2":                             │
│ ❱ 305 │   │   │   raise ValueError("GPT2 models are not supported for merging LORA layers")      │
│   306 │   │                                                                                      │
│   307 │   │   if getattr(self.model, "is_loaded_in_8bit", False):                                │
│   308 │   │   │   raise ValueError("Cannot merge LORA layers when the model is loaded in 8-bit   │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: GPT2 models are not supported for merging LORA layers

**Comment by Mayhem50** [comment]
Hi,

Can we have an update on this topic ?
At least some informations about how to implement it.

Thanks

**Comment by EvenLauvrak** [comment]
Would be great with an update 👍

**Comment by steindor** [comment]
I would appreciate this functionality as well.

**Comment by younesbelkada** [comment]
Hi everyone,
Thanks to @BenjaminBossan , this should be fixed by #676

**Comment by younesbelkada** [comment]
Hi everyone,
If you install PEFT from source:
```bash
pip install git+https://github.com/huggingface/peft.git
```
You should be able to merge and unload GPT2 models
